packages/plugin-ext/src/main/browser/view/views-container-widget.ts

```diff
@@ -0,0 +1,149 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { ViewContainer, View } from '../../../common';
+import { TreeViewWidget } from './tree-views-main';
+import { BaseWidget } from '@theia/core/lib/browser';
+
+export function createElement(className?: string): HTMLDivElement {
+    const div = document.createElement('div');
+    if (className) {
+        div.classList.add(className);
+    }
+    return div;
+}
+
+export interface SectionParams {
+    view: View,
+    container: ViewsContainerWidget
+}
+
+export class ViewsContainerWidget extends BaseWidget {
```

Comment:
I'm adding the view container to the core with proper support of widget lifecycle, toolbars and agnostic to trees: https://github.com/theia-ide/theia/blob/413b39517e574f830ea907de26455350918aaa0a/packages/core/src/browser/view-container.tsx#L21

It would be nice if we sync our effort after both PRs are merged.

Reply:
Sure. I have already seen your PR and pretty interested in use `ViewContainer` from core.